File: llm_generated_code/ChatGPT/Python/Level1/Q6.py
def find_liars(claims):
    n = len(claims)
    
    # Sort the claims
    claims.sort()

    for liars in range(n + 1):
        count = 0
        for claim in claims:
            if claim >= liars:
                count += 1
        if count == liars:
            return liars
    
    return -1  # Contradiction

def main():
    t = int(input())
    results = []
    for _ in range(t):
        n = int(input())
        claims = list(map(int, input().split()))
        result = find_liars(claims)
        if result != -1:
            results.append(result)
        else:
            results.append("Contradiction")
    
    for result in results:
        print(result)

if __name__ == "__main__":
    main()

File: llm_generated_code/ChatGPT/CPP/Level1/Q6.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

// Function to determine the number of liars
int findLiars(vector<int>& claims) {
    int n = claims.size();
    
    // Sort the claims
    sort(claims.begin(), claims.end());

    for (int liars = 0; liars <= n; ++liars) {
        int count = 0;
        for (int i = 0; i < n; ++i) {
            if (claims[i] >= liars) {
                ++count;
            }
        }
        if (count == liars) {
            return liars;
        }
    }
    
    return -1; // Contradiction
}

// Main function to handle multiple test cases
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> claims(n);
        for (int i = 0; i < n; ++i) {
            cin >> claims[i];
        }
        int result = findLiars(claims);
        if (result != -1) {
            cout << result << endl;
        } else {
            cout << "Contradiction" << endl;
        }
    }
    return 0;
}
